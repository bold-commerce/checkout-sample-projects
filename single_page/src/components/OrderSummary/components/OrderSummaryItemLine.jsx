import { Price } from '@boldcommerce/stacks-ui';
import React from 'react';
import RemoveIcon from './RemoveIcon';

const OrderSummaryItemLine = ({ description, amount, onRemove }) => (
  <div className="OrderSummaryItem__Description">
    { onRemove && (
      <button className="OrderSummaryItem__Action" type="button" onClick={onRemove}><RemoveIcon /></button>
    )}
    <span className="OrderSummaryItem__Label">{description}</span>
    <span className="OrderSummaryItem__Amount">(<Price amount={amount}/>)</span>
  </div>
);

export default OrderSummaryItemLine;
