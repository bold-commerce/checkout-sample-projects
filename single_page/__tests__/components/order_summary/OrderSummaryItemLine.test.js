import React from 'react';
import { render } from '@testing-library/react'
import OrderSummaryItemLine from '../../../client/components/OrderSummary/components/OrderSummaryItemLine'

describe('OrderSummaryItemLine', () => {
  test('renders OrderSummaryItemLine', () => {
    const { asFragment } = render(
      <OrderSummaryItemLine
        description={'Test description'}
        amount={1919}
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })

  test('renders OrderSummaryItemLine as removable ', () => {
    const { asFragment } = render(
      <OrderSummaryItemLine
        description={'Test description'}
        amount={1919}
        onRemove={() => {}}
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })
})
