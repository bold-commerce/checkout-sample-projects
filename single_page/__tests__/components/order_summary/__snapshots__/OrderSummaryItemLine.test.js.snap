// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummaryItemLine renders OrderSummaryItemLine 1`] = `
<DocumentFragment>
  <div
    class="OrderSummaryItem__Description"
  >
    <span
      class="OrderSummaryItem__Label"
    >
      Test description
    </span>
    <span
      class="OrderSummaryItem__Amount"
    >
      (
      <div
        class="stx-price stx-price--alight-center"
      >
        <p
          aria-label="Product price"
          class="stx-price__amount"
        >
          $19.19
        </p>
      </div>
      )
    </span>
  </div>
</DocumentFragment>
`;

exports[`OrderSummaryItemLine renders OrderSummaryItemLine as removable  1`] = `
<DocumentFragment>
  <div
    class="OrderSummaryItem__Description"
  >
    <button
      class="OrderSummaryItem__Action"
      type="button"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 13 12"
        width="13"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.14551 12C9.45922 12 12.1455 9.31371 12.1455 6C12.1455 2.68629 9.45922 0 6.14551 0C2.8318 0 0.145508 2.68629 0.145508 6C0.145508 9.31371 2.8318 12 6.14551 12ZM9.30831 2.80811C9.50357 3.00338 9.50357 3.31996 9.30831 3.51522L6.83789 5.98564L9.32005 8.4678C9.51531 8.66306 9.51531 8.97964 9.32005 9.17491C9.12479 9.37017 8.80821 9.37017 8.61294 9.17491L6.13078 6.69275L3.65707 9.16646C3.4618 9.36172 3.14522 9.36172 2.94996 9.16646C2.7547 8.9712 2.7547 8.65462 2.94996 8.45936L5.42368 5.98564L2.95104 3.513C2.75577 3.31774 2.75577 3.00116 2.95104 2.80589C3.1463 2.61063 3.46288 2.61063 3.65814 2.80589L6.13078 5.27853L8.6012 2.80811C8.79646 2.61285 9.11305 2.61285 9.30831 2.80811Z"
          fill="#666666"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <span
      class="OrderSummaryItem__Label"
    >
      Test description
    </span>
    <span
      class="OrderSummaryItem__Amount"
    >
      (
      <div
        class="stx-price stx-price--alight-center"
      >
        <p
          aria-label="Product price"
          class="stx-price__amount"
        >
          $19.19
        </p>
      </div>
      )
    </span>
  </div>
</DocumentFragment>
`;
